### PR TITLE
Feature 5409/Add UI/Reducers for Comments and Bookmark

### PR DESCRIPTION
**Pull request by PhotoNomad0**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- Added reducers/actions for Comments and Bookmark
- Added comments edit dialog
- organized groupmenu actions/helpers
- cleaned out explicit calls to dispatch in Api.js 
- refactored IconIndicators to add click actions and granular control over which icons are shown
- added unit tests for Container.
- this addresses https://github.com/unfoldingword/translationcore/issues/5409 and https://github.com/unfoldingword/translationcore/issues/5410

#### Please include detailed Test instructions for your pull request:
- use test build attached below
- open project in WA
- test comments:
  - verses without comments should show comments icon as white: 
Screen Shot 2020-01-14 at 2 57 03 PM
  - click on comment icon and should see a comment dialog.  Save button should not be enabled until comment has changed.  Enter a comment and click save.  SHould see icon color change to yellow: 
Screen Shot 2020-01-14 at 3 01 03 PM
- test bookmarks:
  - verses without bookmarks should show bookmarks icon as white: 
Screen Shot 2020-01-14 at 3 04 37 PM
  - click on bookmarks icon and should see icon color change to yellow: 
Screen Shot 2020-01-14 at 3 04 44 PM
- exit out of tCore and then restart
- open previous project and should see bookmark and comments icon as yellow.  Click on the bookmark icon and should see your previous comment.

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

### CODE REVIEW

**Comment by PhotoNomad0** [comment]
[Linux build 2.1.0 (ce25085)](https://tc-builds.door43.org/build/feature-mcleanb-5409/translationCore-linux-x64-2.1.0-ce25085.deb) is ready for download.
[macOS build 2.1.0 (ce25085)](https://tc-builds.door43.org/build/feature-mcleanb-5409/translationCore-macos-x64-2.1.0-ce25085.dmg) is ready for download.
[Windows x64 build 2.1.0 (ce25085)](https://tc-builds.door43.org/build/feature-mcleanb-5409/translationCore-win-x64-2.1.0-ce25085.setup.exe) is ready for download.
[Windows x32 build 2.1.0 (ce25085)](https://tc-builds.door43.org/build/feature-mcleanb-5409/translationCore-win-x32-2.1.0-ce25085.setup.exe) is ready for download.

File: src/components/CommentsDialog.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import React from 'react';
```
Comment (mannycolon) [inline src/components/CommentsDialog.js]: Shouldn't this component go into the tc-ui-toolkit?

File: src/state/actions/RemindersActions.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import * as consts from '../actions/actionTypes';
```
Comment (mannycolon) [inline src/state/actions/RemindersActions.js]: Please rename to BookmarkActions

File: src/state/reducers/index.js [hunk 3]
```diff
@@ -2,11 +2,15 @@ import {combineReducers} from 'redux';
...
+import commentsReducer, * as fromCommentsReducer from "./commentsReducer";
+import remindersReducer, * as fromRemindersReducer from "./remindersReducer";
 
 export default combineReducers({
   alignments,
   checks,
-  groupMenu
+  groupMenu,
+  commentsReducer,
+  remindersReducer,
```
Comment (mannycolon) [inline src/state/reducers/index.js]: Rename to bookmarksReducer

File: src/state/reducers/remindersReducer.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import * as consts from '../actions/actionTypes';
```
Comment (mannycolon) [inline src/state/reducers/remindersReducer.js]: Rename to bookmarksReducer

File: src/components/Container.js [hunk 5]
```diff
@@ -539,6 +547,56 @@ class Container extends Component {
...
+  handleBookmarkClick() {
+    const {
+      addReminder,
+      tool: {
+        api
+      },
+      tc: {contextId},
+      username,
+    } = this.props;
+    const bookmarked = getCurrentReminders(api.context.store.getState());
```
Comment (mannycolon) [inline src/components/Container.js]: Why are you accessing the state trough the context and directly from redux as props?

Comment (PhotoNomad0) [reply src/components/Container.js]: Good question, should have mapped state to props

**Pull request merged**